Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts

```diff
@@ -641,6 +660,136 @@ namespace gdjs {
       return false;
     }
 
+    /**
+     * Find the highest floor reachable and move the owner on top of it.
+     * @param candidates The platform to be tested for collision
+     * @param upwardDeltaY The owner won't move upward more than this value.
+     * @param downwardDeltaY The owner won't move downward more than this value.
+     */
+    _findHighestFloorAndMoveOnTop(
+      candidates: gdjs.PlatformRuntimeBehavior[],
+      upwardDeltaY: float,
+      downwardDeltaY: float
+    ) {
+      let totalHighestY = Number.MAX_VALUE;
+      let highestGround: gdjs.PlatformRuntimeBehavior | null = null;
+      for (const platform of candidates) {
+        if (
+          platform.getPlatformType() === gdjs.PlatformRuntimeBehavior.LADDER &&
+          !this._isIn(this._overlappedJumpThru, platform.owner.id)
+        ) {
+          continue;
+        }
+
+        let highestY = this._findPlatformHighestYUnderObject(
+          platform,
+          upwardDeltaY,
+          downwardDeltaY
+        );
+        if (highestY < totalHighestY) {
+          totalHighestY = highestY;
+          highestGround = platform;
+        }
+      }
+      if (highestGround) {
+        const object = this.owner;
+        object.setY(object.getY() + totalHighestY - object.getAABB().max[1]);
+      }
+      return highestGround;
+    }
+
+    /**
+     * Find the highest Y of the floor reachable by the owner.
+     * @param platform The platform to be tested for collision.
+     * @param upwardDeltaY The owner won't move upward more than this value.
+     * @param downwardDeltaY The owner won't move downward more than this value.
+     */
+    private _findPlatformHighestYUnderObject(
+      platform: gdjs.PlatformRuntimeBehavior,
+      upwardDeltaY: float,
+      downwardDeltaY: float
+    ) {
+      const aabb = this.owner.getAABB();
```

Comment:
The previous implementation used collision conditions everywhere for the object, so hitboxes of the platformer object were respected, at least in theory. Here, it seems that the logic is based only on the AABB of the platformer object (but, for platforms, it's using hitboxes properly), so this will be a change right? (not sure if it's a bad thing or not, or if we an make things more precise on demand with hitboxes).

Reply:
Yes, I always forget and think AABB as hitbox AABB. I only realize it doing a small example. For compatibility sake, we could use the hitbox vertices to calculate the min and max every time. Most objects will only have 4 vertices, so it makes 20 compare, jump and affectation.

Reply:
Just to chime in here, based on what I've seen on the discord, a _lot_ of users use way more than 4 vertices on their platform hitboxes, especially when doing ramps or hilly terrain. It is probably crucial to keep those accurate to the hitboxes, and any (realistic) performance difference is going to be felt by a lot of users.

Reply:
It was about the character, but it now use the hitbox AABB instead of the sprite AABB. I think it should be ok to use a rectangular shape for the character.
With this PR, ramps are exactly followed instead of pixel by pixel.

Reply:
Thanks for the clarification! Makes sense to me

Reply:
That's indeed the most impactful change of this PR. I'm hoping that the AABB of the hitboxes of the character is more than enough, but that's something that we'll have to try in the wild with a testing version I think.
This PR is bringing a lot of improvements so we'll definitely want to make sure we can use it and sooner than later :)